lib/mail_room/arbitration/redis.rb

```diff
@@ -0,0 +1,57 @@
+require "redis"
+
+module MailRoom
+  module Arbitration
+    class Redis
+      Options = Struct.new(:redis_url, :namespace) do
+        def initialize(mailbox)
+          redis_url = mailbox.arbitration_options[:redis_url] || "redis://localhost:6379"
+          namespace = mailbox.arbitration_options[:namespace]
+
+          super(redis_url, namespace)
+        end
+      end
+
+      attr_accessor :options
+
+      # Build a new delivery, hold the mailbox configuration
+      # @param [MailRoom::Delivery::Sidekiq::Options]
+      def initialize(options)
+        @options = options
+      end
+
+      def deliver?(message)
+        uid = message.attr["UID"]
+        key = "delivered:#{uid}"
+
+        incr = nil
+        redis.multi do |client|
+          # At this point, `incr` is a future, which will get its value after 
+          # the MULTI command returns.
+          incr = client.incr(key)
+
+          # Expire after 1 minute so Redis doesn't get filled up with outdated data.
+          client.expire(key, 60)
```

Comment:
@DouweM isn't this about races between different mail_room processes, running on different machines? Any one of those machines/processes could hang for minutes.

If one of these locks uses 40 bytes of memory (just making that up) and one receives 1000 emails/minute, it would cost 1000*40*10 = 400000 bytes to keep locks for 10 minutes. I am probably cutting too many corners here but this sounds manageable.